File: PrincipiosDeProgramacionProjects/Cprojects/5.buscarNenM.c
//5.buscarNenM.c
//23 de noviembre de 2023
//
//Este programa lee un numero N de un conjunto de M numeros enteros (M <= 10,000 en el rango de -20,000 hasta 20,000)
//

#include <stdio.h>

int main() {
  int N, M;  //numeros enteros N y M
  int A[10000];  //un array A donde puedo almacenar hasta 10,000 numeros

  //Dar las instrucciones al usuario

  printf("Este programa te indica en que posicion esta el numero (N) en una serie de numeros que dara el usuario")
  printf("NOTA: El usuario debera ingresar los numeros de la lista donde se buscara manera ordenada y ascendente")

  // Leer N
  printf("Ingrese un número a buscar (N): ");
  scanf("%d", &N);

  // Leer M
  printf("Ingrese la cantidad de números que tendra la lista en la que buscaremos el numero N: ");
  scanf("%d", &M);

  // Leer A
  printf("Escriba uno a uno y de manera ordenada los "%d" numeros de la lista", &M)
  for (int i = 0; i < M; i++) {              //Un for para ir solicitando los numeros de la lista a buscar
    printf("Ingrese el número [%d]: ", i);
    scanf("%d", &A[i]);                      // Leer iterativamente 
  }

  // Buscar N en el array A

  int i = 0;                    //Inicializo mi variable
  while (i < M && A[i] < N) {   //mientras i sea menor a M y el valor iesimo de A sea menor a N sigue buscando
    i++;
  }

  // Si N se encontró en A
  if (i < M && A[i] == N) {      //Si el indice es menor al numero de numeros y el valor del iesimo numero de A es igual a N
    printf("N se encuentra en la posición %d de A.\n", i); //Imprimimos donde lo entontro
  } else {
    printf("N no se encuentra en A.\n");
  }

  return 0;
}

File: PrincipiosDeProgramacionProjects/Perlprojects/5.buscarNenM.pl
#5.buscarNenM.pl
#
#Este programa busca en que posicion se encuentra un numero N dentro de un arreglo de M numeros
#paulinapglz.99@gmail.com

#!/usr/bin/perl

use strict;
use warnings;

#Declaramos variables

my $N;    
my $M;
my @A;    #este es el arreglo donde se van a guardar los M numeros

# Leer N
print "Ingrese un numero N que el programa buscara en un conjunto de numeros: ";
chomp($N = <STDIN>);

# Leer M
print "Ingrese la cantidad de números (M) de la lista en la que vamos a buscar: ";
chomp($M = <STDIN>);

# Leer A

for (my $i = 0; $i < $M; $i++) {              #desde 0 hasta menor que el numero de numeros
  print "Ingrese el número A[$i]: ";          #ingresar uno a uno
  chomp($A[$i] = <STDIN>);                    #para el input
}

# Buscar N en A
my $i = 0;                                   #inicializo la variable para empezar a buscar
while ($i < $M && $A[$i] < $N) {             #mientras i sea menor que M y el iesimo numero sea menor a N
  $i++;                                      #sigue buscando
}

#mensaje de exito o fracaso :C (como yo haciendo el de redes)

if ($i < $M && $A[$i] == $N) {                           # Si N se encontró en A
  print "N se encuentra en la posición $i de A.\n";      #se imprime
} else {
  print "N no se encuentra en A.\n";                     #o
}


#done